New package: py-networkit (6.1, 7.0)

fabratu opened the pull request:
A follow-up to #14113. This is the Python-part, while the underlying CPP-lib is already part of spack #15714. The previous suggestions are part of this package.

fabratu committed:
New package py-networkit (6.1, 7.0)

adamjstewart commented on var/spack/repos/builtin/packages/py-networkit/package.py:
```diff
@@ -0,0 +1,45 @@
+    depends_on('py-numpy', type=('build', 'run'))
+    depends_on('py-scipy', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
+    depends_on('py-wheel', type='build')
```
Is wheel actually needed?

adamjstewart commented on var/spack/repos/builtin/packages/py-networkit/package.py:
```diff
@@ -0,0 +1,45 @@
+
+    def build_ext_args(self, spec, prefix):
+        rpath = self.rpath
+        rpath.append(spec['libnetworkit'].prefix.lib)
```
Is this directory not already in `self.rpath`?

fabratu replied:
No, it is not part of the rpath, but directly shared via `LD_LIBRARY_PATH`. When I started the first version of the package some time ago, this was not the case actually. The same goes for the dependency of `wheel`. Both are now deleted.

adamjstewart reviewed (commented):
It looks like there's a way to build in parallel using the `--jobs` arg or `NETWORKIT_PARALLEL_JOBS` environment variable? We should use that to speed up the build.

I think a dependency on `llvm-openmp` is needed when building with Apple Clang. @alalazo is currently working on a PR to split `%clang` and `%apple-clang` into separate compilers. Once that's done, you can add:
```python
depends_on('llvm-openmp', when='%apple-clang')
```

fabratu commented:
Thanks for the input. I adjusted the package accordingly and added parallel building. If I understand the `llvm-openmp` part correctly, the integration/adaptation has to wait for the other PR to finish?

alalazo commented:
> Thanks for the input. I adjusted the package accordingly and added parallel building. If I understand the llvm-openmp part correctly, the integration/adaptation has to wait for the other PR to finish?

@fabratu Feel free to go ahead with that. #17110 has been merged yesterday.
